test/integration/Storage.test.js

```diff
@@ -252,4 +252,22 @@ describe.each([false, true])('Storage (isBatching=%s)', (isBatching) => {
 
         expect(results).toEqual([msg1, msg2, msg3, msg4])
     })
+
+    test('periodically fetch messages in a "recent" range for a particular publisher, msgChainId (not already stored)', async () => {
+        const msg1 = buildEncryptedMsg(streamId, 10, 2000, 0, 'publisher1')
+        // will query periodically until getting some results once the message is stored
+        const streamingResults = storage.requestRange(streamId, 10, 1500, 3, Date.now(), 2, 'publisher1', '1')
+        setTimeout(() => storage.store(msg1), 5000)
```

Comment:
Maybe less than 5000 ms wait is enough to validate this feature, perhaps e.g. 2000 ms so that one / two attempts have passed?